### ISSUE TITLE
JRI not found on Mac

### ISSUE THREAD

**Issue by ngehlenborg** [Caleydo/caleydo#866]
To reproduce:
1. Open with Sample TCGA Data Set
2. Open Tour Guide
3. Add mRNA
4. From mRNA, select the 4 cluster CNMF version and add to StratomeX
5. Click on header brick to select column
6. Right-click and select "Score column"

The following exception is thrown and Caleydo exits:

Cannot find JRI native library!
Please make sure that the JRI native library is in a directory listed in java.library.path.
{{{
java.lang.UnsatisfiedLinkError: /Users/nils/Projects/Juno/org.caleydo.util.r/lib/macosx/libjri.jnilib: dlopen(/Users/nils/Projects/Juno/org.caleydo.util.r/lib/macosx/libjri.jnilib, 1): Library not loaded: /Library/Frameworks/R.framework/Versions/2.13/Resources/lib/libR.dylib
  Referenced from: /Users/nils/Projects/Juno/org.caleydo.util.r/lib/macosx/libjri.jnilib
  Reason: image not found
    at java.lang.ClassLoader$NativeLibrary.load(Native Method)
    at java.lang.ClassLoader.loadLibrary1(ClassLoader.java:1939)
    at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1864)
    at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1838)
    at java.lang.Runtime.loadLibrary0(Runtime.java:845)
    at java.lang.System.loadLibrary(System.java:1084)
    at org.rosuda.JRI.Rengine.<clinit>(Rengine.java:19)
    at org.caleydo.util.r.RStatisticsPerformer.init(RStatisticsPerformer.java:84)
    at org.caleydo.util.r.RStatisticsPerformer.<init>(RStatisticsPerformer.java:67)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:525)
    at java.lang.Class.newInstance0(Class.java:372)
    at java.lang.Class.newInstance(Class.java:325)
    at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:184)
    at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:905)
    at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:243)
    at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:55)
    at org.caleydo.core.manager.GeneralManager.getRStatisticsPerformer(GeneralManager.java:197)
    at org.caleydo.core.data.perspective.table.AdjustedRandIndex.getScore(AdjustedRandIndex.java:60)
    at org.caleydo.view.tourguide.data.score.AdjustedRankScore.apply(AdjustedRankScore.java:45)
    at org.caleydo.view.tourguide.data.ScoreQuery.compute(ScoreQuery.java:95)
    at org.caleydo.view.tourguide.data.ScoreQuery.apply(ScoreQuery.java:72)
    at org.caleydo.view.tourguide.vendingmachine.VendingMachine.recomputeScores(VendingMachine.java:340)
    at org.caleydo.view.tourguide.vendingmachine.VendingMachine.access$2(VendingMachine.java:336)
    at org.caleydo.view.tourguide.vendingmachine.VendingMachine$3.propertyChange(VendingMachine.java:130)
    at java.beans.PropertyChangeSupport.fire(PropertyChangeSupport.java:335)
    at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:328)
    at java.beans.PropertyChangeSupport.fireIndexedPropertyChange(PropertyChangeSupport.java:358)
    at org.caleydo.view.tourguide.data.ScoreQuery.addSelection(ScoreQuery.java:241)
    at org.caleydo.view.tourguide.vendingmachine.VendingMachine.onAddColumn(VendingMachine.java:401)
    at org.caleydo.view.tourguide.vendingmachine.VendingMachine.createStratificationScore(VendingMachine.java:249)
    at org.caleydo.view.tourguide.listener.ScoreTablePerspectiveListener.handleEvent(ScoreTablePerspectiveListener.java:45)
    at org.caleydo.core.view.opengl.canvas.AGLView.processEvents(AGLView.java:526)
    at org.caleydo.core.view.opengl.canvas.AGLView.display(AGLView.java:330)
    at org.caleydo.core.view.opengl.canvas.internal.swt.SWTGLCanvas.fireDisplay(SWTGLCanvas.java:116)
    at org.caleydo.core.view.opengl.canvas.internal.swt.SWTGLCanvas$2.display(SWTGLCanvas.java:96)
    at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:373)
    at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:358)
    at com.jogamp.opengl.swt.GLCanvas$2.run(GLCanvas.java:137)
    at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:655)
    at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:594)
    at com.jogamp.opengl.swt.GLCanvas$3.run(GLCanvas.java:148)
    at org.eclipse.swt.internal.cocoa.OS.objc_msgSend_bool(Native Method)
    at org.eclipse.swt.internal.cocoa.NSRunLoop.runMode(NSRunLoop.java:42)
    at org.eclipse.swt.widgets.Display.sleep(Display.java:4565)
    at org.eclipse.ui.application.WorkbenchAdvisor.eventLoopIdle(WorkbenchAdvisor.java:364)
    at org.eclipse.ui.internal.Workbench$3.eventLoopIdle(Workbench.java:462)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$9.run(PartRenderingEngine.java:1033)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:923)
    at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:86)
    at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:588)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:543)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
    at org.caleydo.core.startup.StartupProcessor.initRCPWorkbench(StartupProcessor.java:228)
    at org.caleydo.core.startup.StartupProcessor.initStartupProcudure(StartupProcessor.java:95)
    at org.caleydo.core.startup.Application.start(Application.java:44)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:353)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:180)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:629)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:584)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1438)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1414)
}}}

**Comment by ngehlenborg** [comment]
I noticed that the JRI subdirectories for Linux and Windows contain architecture subfolders, which is not the case for Mac.

**Comment by ngehlenborg** [comment]
Adding architecture "x84-64" to the manifest didn't help, nor did installing rJava in R.

**Comment by sgratzl** [comment]
the jri library seems to be highly version dependent on mac. In this case if you take a look at the output, you see that it tries to load `/Library/Frameworks/R.framework/Versions/2.13/Resources/lib/libR.dylib` -> which is somewhere hardcoded in the library. 

-> it depends which version you have on the system.

using following steps it runs on my system: 
1. install R
2. R console: `install.packages("rJava")` `install.packages("clues")`
3. R console: `system.file("jri",package="rJava")`
4. open in finder the resolved path and copy the files `JRI.jar` and `libjri.jnilib` to `org.caleydo.util.r/lib`respectively `org.caleydo.util.r/lib/macosx`
5. Update within the launch configuration -> Environment -> R_HOME to `/Library/Frameworks/R.framework/Resources` to prevent errors like `cannot find system Renviron`
6. then it should work

This is not a real solution to the problem but a workaround

**Comment by alexsb** [comment]
Depends on #1132

**Comment by sgratzl** [comment]
if we implement #1132 in a general, we don't need JRI anymore within Caleydo. And otherwise I would suggest to dynamically load the JRI library

-> wontfix
